integration/query_comparision_test.go

```diff
@@ -545,8 +545,39 @@ func TestQueryComparisonLte(t *testing.T) {
 	}
 }
 
+func TestQueryComparisonNin(t *testing.T) {
+	t.Parallel()
+	providers := []shareddata.Provider{shareddata.Scalars, shareddata.Composites}
+	ctx, collection := setup(t, providers...)
+
+	var docValue bson.A
+	for _, expected := range shareddata.Scalars.Docs() {
+		docValue = append(docValue, expected.Map()["value"])
+	}
+
+	for name, tc := range map[string]struct {
+		q           bson.D
+		expectedIDs []any
+	}{
```

Comment:
Could you please add tests for bad values like `nil` or `string` and for empty array?

Reply:
Do you mean like another seperate test cases? 
`docValue` variable has all scalar data types such as `string`, `nil`, etc. The variable doesn't have composite data type though. So yeah, if need to write tests for composite data type, I'll do it.

Reply:
Yeah, I mean a query like `{"$nin", nil}` or `{"$nin", "some-string"}`

Reply:
done